### PR TITLE
Added function service and unit test

### PR DESCRIPTION
**Pull request by thoughtpalette**
This PR adds the following functionality:
- Adding Function Rest API to nClient
- Adds Unit tests for newly added Functions Rest API
- Updates package json version
- Updates imports/exports where needed.

---

Screenshot 2023-01-25 at 2 07 24 PM

Screenshot 2023-01-25 at 2 08 03 PM

### CODE REVIEW

File: src/client/functions/FunctionService.ts [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+				...this.#serviceConfig.getDefaultOptions(),
+			},
+			this.#serviceConfig.fetch
+		)
+
+		if (functions.isEmpty()) {
+			return []
+		}
+
+		return functions.getRows()
```
Comment (hecsalazarf) [inline src/client/functions/FunctionService.ts]: Maybe ternary operator?

File: src/client/functions/FunctionService.ts [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+	public async readById(id: string | HRef): Promise<FunctionType> {
+		const finFunction = await fetchVal<FunctionType>(
+			`${this.#url}/${HRef.make(id).value}`,
+			{
+				...this.#serviceConfig.getDefaultOptions(),
+			},
+			this.#serviceConfig.fetch
+		)
+
+		return finFunction
```
Comment (hecsalazarf) [inline src/client/functions/FunctionService.ts]: Do you think `fetchVal` could be directly returned?

**Review by hecsalazarf — APPROVED** [review]
LGTM

File: src/client/functions/FunctionService.ts (hunk 2 above)
Comment (thoughtpalette) [reply src/client/functions/FunctionService.ts]: Def! Added.

File: src/client/functions/FunctionService.ts (hunk 1 above)
Comment (thoughtpalette) [reply src/client/functions/FunctionService.ts]: Good suggestion! Added.

**Pull request merged**